packages/announcement-controller/src/AnnouncementController.ts

```diff
@@ -1,96 +1,116 @@
+import type { Patch } from 'immer';
 import {
-  BaseController,
-  BaseConfig,
-  BaseState,
+  BaseControllerV2,
+  RestrictedControllerMessenger,
 } from '@metamask/base-controller';
 
-interface ViewedAnnouncement {
+type ViewedAnnouncement = {
   [id: number]: boolean;
-}
+};
 
-interface Announcement {
+type Announcement = {
   id: number;
   date: string;
-}
-
-interface StateAnnouncement extends Announcement {
-  isShown: boolean;
-}
+};
 
-/**
- * A map of announcement ids to Announcement objects
- */
-interface AnnouncementMap {
-  [id: number]: Announcement;
-}
+type StateAnnouncement = Announcement & { isShown: boolean };
 
 /**
  * A map of announcement ids to StateAnnouncement objects
  */
-export interface StateAnnouncementMap {
+export type StateAnnouncementMap = {
   [id: number]: StateAnnouncement;
-}
-
-/**
- * AnnouncementConfig will hold the active announcements
- */
-export interface AnnouncementConfig extends BaseConfig {
-  allAnnouncements: AnnouncementMap;
-}
+};
 
 /**
  * Announcement state will hold all the seen and unseen announcements
  * that are still active
  */
-export interface AnnouncementState extends BaseState {
+export type AnnouncementState = {
   announcements: StateAnnouncementMap;
-}
+};
+
+export type AnnouncementControllerActions = GetAnnouncementState;
+export type AnnouncementControllerEvents = AnnouncementStateChange;
+
+export type GetAnnouncementState = {
+  type: `${typeof controllerName}:getState`;
+  handler: () => AnnouncementState;
+};
+
+export type AnnouncementStateChange = {
+  type: `${typeof controllerName}:stateChange`;
+  payload: [AnnouncementState, Patch[]];
+};
+
+const controllerName = 'AnnouncementController';
 
 const defaultState = {
   announcements: {},
 };
 
+const metadata = {
+  announcements: {
+    persist: true,
+    anonymous: false,
```

Comment:
I think we can set this to `true`; this state doesn't reveal anything about the user. It just shows whether they've seen the announcements